docs/guides/quarkus.md

```diff
@@ -0,0 +1,326 @@
+say that it is a configuration file that is used when building your project.
+If you're really interested, check out the [reference doc](config-reference.doc).
+To learn more about the CLI and the details for each supported command, see
+the [CLI Commands document](commands.md#cli-commands).
+
+## Running the function locally
+
+To run a function, you'll first need to build it. This step creates an OCI
+container image that can be run locally on your computer, or on a Kubernetes
+cluster.
+
+```
+❯ func build
+```
+You might be asked about docker registry during the build step.
+After the function has been built, it can be run locally.
+
+```
+❯ func run
+```
+
+Functions can be invoked with a simple HTTP request. 
+You can test to see if the function is working by using your browser to visit
+http://localhost:8080.
+
+## Deploying the function to a cluster
+
+To deploy your function to a Kubenetes cluster, use the `deploy` command.
+
+```
+❯ func deploy
+```
+
+You can get the URL for your deployed function with the `describe` command.
+
+```
+❯ func describe
+```
+
+## Testing a function locally
+
+
+Quarkus functions can be tested locally on your computer.
+The project contains tests which you can run.
+You do this as you would with any maven project.
+
+```
+❯ mvn test
+```
+
+## Function reference
+
+Boson Quarkus functions have very few restrictions. You can work with it as with any Java maven project.
+The only real requirements is that your project contain a method annotated with `@Funq`.
+In this section, we will look in a little more detail at how Boson functions are invoked,
+and what APIs are available to you as a developer.
+
+### Invocation parameters
+
+When using the `func` CLI to create a function project, you may choose to generate a project
+that responds to a `CloudEvent` or simple HTTP. `CloudEvents` in Knative are transported over
+HTTP as a `POST` request, so in many ways, the two types of functions are very much the same.
+They each will listen and respond to incoming HTTP events.
+
+When an incoming request is received, your function will be invoked with an instance of type of your choice, see [Types](#types).
+
+The instance will contain one of:
+* `data` of `CloudEvent`
+* Body of HTTP POST request
+* Query parameters of HTTP GET request
+
+For example, if `CloudEvent` contains a JSON object such as this in its data property
+(or if an ordinary HTTP POST that contains such an object in body),
+
+```json
+{
+  "customerId": "0123456",
+  "productId": "6543210"
+}
+```
+
+you might want to use JavaBean similar to this as an input parameter:
+
+```java
+public class Purchase {
+    private long customerId;
+    private long productId;
+    
+    // getters and setter here
+}
+```
+
+and function signature could look like this:
+
+```java
+public class Functions {
+    @Funq
+    public void processPurchase(Purchase purchase) {
+        // do stuff
+    }
+}
+```
+
```

Comment:
```suggestion
For example, imagine a function that is meant to receive and process purchase data.
Your function signature may look like this:

```java
public class Functions {
    @Funq
    public void processPurchase(Purchase purchase) {
        // do stuff
    }
}
```

With a `Purchase` bean that looks like this:

```java
public class Purchase {
    private long customerId;
    private long productId;
    
    // getters and setter here
}
```

Expecting data that may be represented in JSON like this:

```json
{
  "customerId": "0123456",
  "productId": "6543210"
}
```

Depending on the deployment, this function will be invoked in one of three ways:

* An incoming `CloudEvent` with a JSON object such as above in its `data` property. 
* An ordinary HTTP POST with a JSON object such as above in the body of the request
* An ordinary HTTP GET with a query string like `customerId=0123456&productId6543210

```

Reply:
Github formatting is weird here.

Reply:
@lance this suggest to delete a lot, is that what we want?

Reply:
It only looks like it's deleting a lot because of how GH is formatting my suggestion. I'll share it with you elsewhere.

Reply:
Ah I see.